src/OpenTelemetry.Api/Metrics/Timer.cs

```diff
@@ -0,0 +1,169 @@
+// <copyright file="Timer.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Diagnostics.Metrics;
+
+namespace OpenTelemetry.Metrics
+{
+    /// <summary>
+    /// Timer instrument.
+    /// </summary>
+    /// <typeparam name="T">Support <c>int</c>, <c>long</c>, <c>double</c>.</typeparam>
+    public sealed class Timer<T> : Instrument<T>
```

Comment:
I'm concerned about the name `Timer` since it sounds like very generic.

Reply:
Naming is always so hard.

Maybe, some combination of these words/concepts?
Interval, Period, Span
Elapsed, Duration, Time
Timer, Counter, Stopwatch

IntervalTimer? ElapsedDurationCounter? etc...